addons/bus/models/res_partner.py

```diff
@@ -34,24 +34,27 @@ def im_search(self, name, limit=20):
             :param name : the partner name to search
             :param limit : the limit of result to return
         """
-        name = '%' + name + '%'
-        excluded_partner_ids = [self.env.user.partner_id.id]
-        self.env.cr.execute("""
-            SELECT
-                U.id as user_id,
-                P.id as id,
-                P.name as name,
-                CASE WHEN B.last_poll IS NULL THEN 'offline'
-                     WHEN age(now() AT TIME ZONE 'UTC', B.last_poll) > interval %s THEN 'offline'
-                     WHEN age(now() AT TIME ZONE 'UTC', B.last_presence) > interval %s THEN 'away'
-                     ELSE 'online'
-                END as im_status
-            FROM res_users U
-                JOIN res_partner P ON P.id = U.partner_id
-                LEFT JOIN bus_presence B ON B.user_id = U.id
-            WHERE P.name ILIKE %s
-                AND P.id NOT IN %s
-                AND U.active = 't'
-            LIMIT %s
-        """, ("%s seconds" % DISCONNECTION_TIMER, "%s seconds" % AWAY_TIMER, name, tuple(excluded_partner_ids), limit))
-        return self.env.cr.dictfetchall()
+        if self.env['mail.channel'].check_access_rights('create', raise_exception=False):
```

Comment:
So that would be, in the context of channel *creation* or *extension via an invite* - we could say both require the `create` right ;-)